File: app/src/main/java/com/wxm/camerajob/data/define/EProperty.java
package com.wxm.camerajob.data.define;

import android.support.annotation.Nullable;

/**
 * for properties
 * Created by ookoo on 2018/2/21.
 */
public enum EProperty {
    PROPERTIES_FILENAME("appConfig.properties"),
    PROPERTIES_CAMERA_FACE("camera_face"),
    PROPERTIES_CAMERA_DPI("camera_dpi"),
    PROPERTIES_CAMERA_AUTO_FLASH("camera_auto_flash"),
    PROPERTIES_CAMERA_AUTO_FOCUS("camera_auto_focus");

    private String szPropertiesName;

    EProperty(String szName)  {
        szPropertiesName = szName;
    }

    /**
     * get properties name
     * @return  name
     */
    public String getName()  {
        return szPropertiesName;
    }

    /**
     * use name to get enum object
     * @param szName    property name
     * @return      enum object
     */
    @Nullable
    public static EProperty getEPropertyName(String szName) {
        for(EProperty en : EProperty.values())  {
            if(en.szPropertiesName.equals(szName))
                return en;
        }

        return null;
    }
}

File: app/src/main/java/com/wxm/camerajob/data/define/EProperty.kt
package com.wxm.camerajob.data.define

/**
 * for properties
 * Created by WangXM on 2018/2/21.
 */
enum class EProperty(val paraName: String) {
    PROP_FILENAME("appConfig.properties"),
    PROP_CAMERA_FACE("camera_face"),
    PROP_CAMERA_DPI("camera_dpi"),
    PROP_CAMERA_AUTO_FLASH("camera_auto_flash"),
    PROP_CAMERA_AUTO_FOCUS("camera_auto_focus"),

    PROP_CAMERA_CAPTURE_TRY_COUNT("camera_capture_try_count"),
    PROP_CAMERA_CAPTURE_SKIP_FRAME("camera_capture_skip_frame");
}
